### PR TITLE
[ML] handle new model metadata stream from native process

### PR DESCRIPTION
**Pull request by benwtrent**
This adds the serialization handling for the new model_metadata object from the native process. 

Right now, this document only stores total_feature_importance information.

It is also not yet currently consumed by any user consumed API.

relates: https://github.com/elastic/ml-cpp/pull/1387

### CODE REVIEW

File: x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/inference_index_template.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "order" : 0,
   "version" : ${xpack.ml.version.id},
   "index_patterns" : [
-    ".ml-inference-000002"
+    ".ml-inference-000003"
```
Comment (benwtrent) [inline x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/inference_index_template.json:5]: Are these templates automatically installed when any new node comes online? Or does the new node have to be a master node?

@dimitris-athanasiou do you know?

Comment (dimitris-athanasiou) [reply x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/inference_index_template.json:5]: Aside of the question, I believe the index pattern here should be `.ml-inference-*` and we shouldn't have to change this each time.

Back to the question, all nodes get the templates installed regardless of whether they're master or not.

Comment (benwtrent) [reply x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/inference_index_template.json:5]: @dimitris-athanasiou I am not sure. What if an older node has a specific mapping in mind and tries to write to the index?

A new index would be created with the new mapping (if it didn't already exist) and the new node does not have code to support the new mapping. 

Keeping it versioned allows only NEW nodes to reference the new index mapping.

Comment (dimitris-athanasiou) [reply x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/inference_index_template.json:5]: I see, it makes sense.

**Comment by davidkyle** [comment]
The test failures are 
 
> {"type":"illegal_argument_exception","reason":"Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [model_id]

**model_id** is missing from the mappings in the template

**Review by davidkyle — COMMENTED** [review]
Looks good, no comments just the test failures to fix

**Comment by benwtrent** [comment]
run elasticsearch-ci/2

**Comment by benwtrent** [comment]
run elasticsearch-ci/2

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**